### PR TITLE
[FrameworkBundle] Add `--resolve-env-vars` option to `lint:container` command

### LINKED ISSUE #58105: [FrameworkBundle] lint:container should check if referenced env variables exist
### Description

It's common mistake in our teams to define environment variable for one environment (eg. staging), but not the other (eg. production). Deploying such application with such configuration works and even most of the routes, but once one of the route needs service that requires the env var, at that point things blow up with this error

```
Noticed exception 'Symfony\Component\DependencyInjection\Exception\EnvNotFoundException' with message 'Environment variable not found: "PM_DC_TOKEN".' in /srv/api/vendor/symfony/dependency-injection/EnvVarProcessor.php:221
```

So I thought one neat way how I can avoid such issues in future is to add CI job which runs `bin/console lint:container` - both during CI run and during deployment, however this command does not complain about this. I'm wondering what else I can do?

symfony version I use: 7.1.3

### Example

_No response_

### PR DESCRIPTION
**Pull request by ostrolucky**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? |no
| Issues        | Fix #58105
| License       | MIT

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php [hunk 1]
```diff
@@ -39,6 +40,7 @@ protected function configure(): void
     {
         $this
             ->setHelp('This command parses service definitions and ensures that injected values match the type declarations of each services\' class.')
+            ->addOption('resolve-env-vars', null, InputOption::VALUE_NONE, 'When define, command will try to resolve all env variables')
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php]:
```suggestion
            ->addOption('resolve-env-vars', null, InputOption::VALUE_NONE, 'Resolve environment variables and fail if one is missing.')
```

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php [hunk 2]
```diff
@@ -58,7 +60,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $container->setParameter('container.build_time', time());
 
         try {
-            $container->compile();
+            $container->compile((bool) $input->getOption('resolve-env-vars'));
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerLintCommand.php:63]:
```suggestion
            $container->compile($input->getOption('resolve-env-vars'));
```

**Review by nicolas-grekas — APPROVED** [review]
LGTM after some minor CS changes (suggestions partly reverted).

**Comment by nicolas-grekas** [comment]
Isn't the failure related?

**Comment by ostrolucky** [comment]
Looks it got resolved with rebase :)

**Comment by nicolas-grekas** [comment]
Thank you @ostrolucky.

**Pull request merged**